samples/attested_tls/server/enc/openssl/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) Open Enclave SDK contributors.
+# Licensed under the MIT License.
+
+# Use the edger8r to generate C bindings from the EDL file.
+add_custom_command(
+  OUTPUT tls_server_t.h tls_server_t.c tls_server_args.h
+  DEPENDS ${CMAKE_SOURCE_DIR}/server/tls_server.edl
+  COMMAND
+    openenclave::oeedger8r --trusted ${CMAKE_SOURCE_DIR}/server/tls_server.edl
+    --search-path ${OE_INCLUDEDIR} --search-path
+    ${OE_INCLUDEDIR}/openenclave/edl/sgx)
```

Comment:
Can we separate SGX-specific things from TEE-agnostic things?
For example you nicely have OE_CRYPTO_LIB in line 17 but here hard-codes SGX instead of using something like OE_TEE

Reply:
Looks like hardcoded sgx value is repeated across all the sample files. Does that mean current samples support only sgx platform?